src/Nethermind/Nethermind.Mev/Data/MevBundle.cs

```diff
@@ -18,21 +18,31 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Threading;
 using Nethermind.Core;
 using Nethermind.Core.Crypto;
 using Nethermind.Int256;
+using Nethermind.Mev.Source;
+using Nethermind.Serialization.Rlp;
 
 namespace Nethermind.Mev.Data
 {
-    public class MevBundle : IEquatable<MevBundle>
+    public class MevBundle : IEquatable<MevBundle>, IComparable<MevBundle>
     {
+        private static int _poolIndex = 0;
+
         public MevBundle(IReadOnlyList<Transaction> transactions, long blockNumber, UInt256? minTimestamp, UInt256? maxTimestamp, Keccak[]? revertingTxHashes = null)
         {
             Transactions = transactions;
             BlockNumber = blockNumber;
+
+            Rlp rlp = Rlp.Encode(Rlp.Encode(BlockNumber), Rlp.Encode(transactions.Select(t => t.Hash).ToArray()));
```

Comment:
remove, simply use GetHashCode, RLP is an unnecessary allocation here.
Even if needed this RLP is done a suboptimal way (allocating a lot of memory)

Reply:
We wanted to introduce something called Bundle identity.

We can rework it to RlpStream.